deployment/params.py

```diff
@@ -27,183 +19,257 @@
+        variable_elements = variable.split(",")
+        method_name = variable_elements[0]
+        method_args = list()
+        if len(variable_elements) > 1:
+            args = variable_elements[1:]
+            for arg in args:
+                processed_value = _process_raw_value(arg, context)
+                method_args.append(processed_value)
+
+        contract_name = context.contract_name
+        contract_container = get_contract_container(contract_name)
+        contract_method_abis = contract_container.contract_type.methods
+        method_abi = None
+        for abi in contract_method_abis:
+            if abi.name == method_name:
+                method_abi = abi
+        if not method_abi:
+            raise ValueError(f"ABI could not be found for method {contract_name}.{method_name}")
+        input_abi_types = [t.type for t in method_abi.inputs]
+        if len(input_abi_types) != len(method_args):
+            raise ValueError(
+                f"{contract_name}.{method_name} parameters length mismatch - "
+                f"ABI requires {len(input_abi_types)}, Got {len(method_args)}."
+            )
+
+        return method_name, input_abi_types, method_args
+
+    @classmethod
+    def is_encode(cls, value: str) -> bool:
+        """Returns True if the variable is a variable that needs encoding to bytes"""
+        return value.startswith(cls.ENCODE_PREFIX)
+
+    def resolve(self) -> Any:
+        contract_container = get_contract_container(self.contract_name)
+        contract_instance = _get_contract_instance(contract_container)
+        if contract_instance == ZERO_ADDRESS:
+            # logic contract not yet deployed - in eager validation check
+            return "0xdeadbeef"  # something noticeable in case ever actually returned
+
+        method_args = list()
+        for method_arg in self.method_args:
+            value = method_arg
+            if isinstance(method_arg, Variable):
+                value = method_arg.resolve()
+            method_args.append(value)
+
+        method_handler = getattr(contract_instance, self.method_name)
+        encoded_bytes = method_handler.encode_input(*method_args)
+        return encoded_bytes.hex()  # return as hex - just cleaner
+
+
+class ContractName(Variable):
+    def __init__(self, contract_name: str, context: VariableContext):
+        if contract_name not in context.contract_names:
+            raise ValueError(f"Contract name {contract_name} not found")
+
+        self.contract_name = contract_name
+        self.check_for_proxy_instances = context.check_for_proxy_instances
+
+    def resolve(self) -> Any:
+        """Resolves a contract address."""
+        contract_container = get_contract_container(self.contract_name)
+        contract_instance = _get_contract_instance(contract_container)
+        if contract_instance == ZERO_ADDRESS:
+            # eager validation
+            return ZERO_ADDRESS
```

Comment:
Should we also return `"0xdeadbeef"` here?

Reply:
This function is expected to return a contract address (`address` type) so `0xdeadbeef` would fail validation. The ZERO_ADDRESS is basically a placeholder for validating referenced contract address (contract variable) during initial eager validation before contracts are actually deployed.